CMakeLists.txt

```diff
@@ -189,8 +189,10 @@ else()
 		# if OPENSSL_ROOT_DIR or USE_OPENSSL_PC are set, we assume
 		# that the user wants to use find_package(OpenSSL) method to
 		# find OpenSSL package. Otherwise default to pkg-config.
-		if (NOT OPENSSL_ROOT_DIR OR USE_OPENSSL_PC)
-			pkg_check_modules(SSL "openssl libcrypto")
+		if (NOT IOS)
```

Comment:
I'm in disfavor of this solution because this fixes the problem on IOS only.

Reply:
Agreed. How about this:
```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index eb2582c..f379416 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,6 +90,7 @@ option(USE_GNUTLS "Should use gnutls instead of openssl" OFF)
 option(ENABLE_CXX_DEPS "Extra library dependencies in srt.pc for the CXX libraries useful with C language" ON)
 option(USE_STATIC_LIBSTDCXX "Should use static rather than shared libstdc++" OFF)
 option(ENABLE_INET_PTON "Set to OFF to prevent usage of inet_pton when building against modern SDKs while still requiring compatibility with older Windows versions, such as Windows XP, Windows Server 2003 etc." ON)
+option(USE_OPENSSL_PC "Use pkg-config to find OpenSSL libraries" ON)
 
 set(TARGET_srt "srt" CACHE STRING "The name for the haisrt library")
 
@@ -186,10 +187,9 @@ if ( USE_GNUTLS )
 else()
                message(STATUS "SSL Dependency: using OpenSSL (default)")
 
-               # if OPENSSL_ROOT_DIR or USE_OPENSSL_PC are set, we assume
-               # that the user wants to use find_package(OpenSSL) method to
-               # find OpenSSL package. Otherwise default to pkg-config.
-               if (NOT OPENSSL_ROOT_DIR OR USE_OPENSSL_PC)
+               # Try using pkg-config method first if enabled,
+               # fall back to find_package method otherwise
+               if (USE_OPENSSL_PC)
                        pkg_check_modules(SSL "openssl libcrypto")
                endif()
                if (SSL_FOUND)
```
@sergeiignatov ?

Reply:
I agree. I updated CMakeLists.txt and iOS build how-to.

Reply:
Makes sense. This should help:
```
diff --git a/.travis.yml b/.travis.yml
index ee447b1..1557945 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -41,13 +41,16 @@ matrix:
             - ./Configure --cross-compile-prefix=x86_64-w64-mingw32- mingw64
             - make
             - cd ..
-            - export OPENSSL_ROOT_DIR="$PWD/openssl"
-            - export CMAKE_SYSTEM_NAME="Windows"
-            - export CC="x86_64-w64-mingw32-gcc"
-            - export CXX="x86_64-w64-mingw32-g++"
           env: BUILD_TYPE=Release
 script:
-    - if [ "$TRAVIS_OS_NAME" == "linux" ]; then cmake . -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DENABLE_UNITTESTS="ON" -DOPENSSL_ROOT_DIR="$OPENSSL_ROOT_DIR" -DCMAKE_SYSTEM_NAME="$CMAKE_SYSTEM_NAME"; fi
-    - if [ "$TRAVIS_OS_NAME" == "osx" ];   then cmake . -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DENABLE_UNITTESTS="ON" -DCMAKE_PREFIX_PATH=/usr/local/opt/openssl; fi
+    - if [ "$TRAVIS_COMPILER" == "x86_64-w64-mingw32-g++" ]; then
+        export CC="x86_64-w64-mingw32-gcc";
+        export CXX="x86_64-w64-mingw32-g++";
+        cmake . -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DENABLE_UNITTESTS="ON" -DUSE_OPENSSL_PC="OFF" -DOPENSSL_ROOT_DIR="$PWD/openssl" -DCMAKE_SYSTEM_NAME="Windows";
+      elif [ "$TRAVIS_OS_NAME" == "linux" ]; then
+        cmake . -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DENABLE_UNITTESTS="ON";
+      elif [ "$TRAVIS_OS_NAME" == "osx" ]; then
+        cmake . -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DENABLE_UNITTESTS="ON" -DCMAKE_PREFIX_PATH=/usr/local/opt/openssl;
+      fi
     - make
     - if [ "$TRAVIS_COMPILER" != "x86_64-w64-mingw32-g++" ]; then ctest --extra-verbose; fi
```
I tested the above change in a fork and the build is passing.

Reply:
@sergeiignatov ,

Is there a chance you can add the above change to your PR so we can get CI to pass?

Reply:
Done.